sidecar/src/main.rs

```diff
@@ -38,7 +38,7 @@ fn parse_query_line(string: &str) -> QueryType {
 }
 
 enum Addrs<'a> {
-    Args(Values<'a>),
+    Args(std::vec::IntoIter<String>),
```

Comment:
This replacement doesn't seem particularly elegant to me. Now you are forced to `take` from `config` below, which is error prone, because subsequent code may expect the addresses to be still available. In my opinion the following would be better:
```diff
--- src/main.rs
+++ src/main.rs
@@ -38,7 +38,7 @@ fn parse_query_line(string: &str) -> QueryType {
 }

 enum Addrs<'a> {
-    Args(std::vec::IntoIter<String>),
+    Args(&'a mut dyn Iterator<Item = &'a str>),
     Stdin(Lines<StdinLock<'a>>),
 }

@@ -432,11 +430,13 @@ fn main() {
     let dwarf = Arc::new(dwarf);
     let ctx = Context::from_arc_dwarf(Arc::clone(&dwarf)).unwrap();

+    let mut addrs;
     let stdin = std::io::stdin();
     let queries = if config.addrs.is_empty() {
         Addrs::Stdin(stdin.lock().lines())
     } else {
-        Addrs::Args(take(&mut config.addrs).into_iter())
+        addrs = config.addrs.iter().map(String::as_ref);
+        Addrs::Args(&mut addrs)
     };

     for addr_or_cunit in queries {
```